### DISCUSSION TITLE
Raw Markup Field Type Displaying in Incoming Form

### DISCUSSION THREAD

**Discussion by amberh1** [sheltermanager/asm3#1587] [Q&A / Help] [answered]
I have attached a screenshot of a test form showing a raw markup type of field. I have code in the raw markup to create the input boxes for first and last name. The input of the form seems to be correctly stored (as seen in the text at the top of the screenshot; Test First Name and test last name). The issue is the raw markup displays on the incoming form. It seems to do that with any raw markup. Is there a reason for this, or is there any way I can prevent this from happening and only have the text information show?
![image](https://github.com/user-attachments/assets/e2b7f420-7084-4129-8f60-53c89917cad0)

**Comment by bobintetley** [comment]
In the main, people use the raw markup fields for adding clauses and extra information/text to the forms - so they want those things in the final form.

Why would you create input fields in your raw markup like this, rather than just making them text fields in the form? Doing it this way, you lose all of our form functionality, like being able to use "showif" to control visibility based on the value of other fields, or being able to make fields mandatory, etc.

You could inject extra elements into the DOM with code in the header if you need to add things that shouldn't appear in the form output at the end.

**Reply by amberh1** [reply]
The form fields on ASM don't seem to have the same flexibility that creating my own fields would have. I'm able to style ASM form fields for the most part (font, size, border, etc.), but I can't figure out how to adjust for more complex styling. For example, I would want first and last name horizontal to each other instead of vertically aligned. I have managed to do that via my code in the raw markup (attached image), and the fields are mandatory, too, and don't allow the user to submit without filling them out. Is it possible to change the alignment with ASM form fields? I noticed via the source code that the fields are in tables instead of div containers, and I'm not sure I'm able to edit the tables.
![image](https://github.com/user-attachments/assets/407bef90-18c2-4b35-b4b6-4d03094eb175)

**Reply by bobintetley** [reply, marked as answer]
Ok, I see. Yes, I'm afraid that is one limitation that can't be worked around easily - the fields have to be laid out vertically due to them being in table structures. We are considering how this might be changed in future.

For now though, if you want to change the layout completely, the only option is to make the form in our system, then make your own version of the form - and use the same names as the input/select fields as ours, and with the hidden inputs from the top of our form. You can then have your form POST to our handler.
